keras_cv/layers/preprocessing/mosaic.py

```diff
@@ -151,10 +151,8 @@ def _batch_augment(self, inputs):
                 axis=-1,
             )
 
-            if isinstance(bounding_boxes, tf.RaggedTensor):
-                bounding_boxes = bounding_boxes.to_tensor(-1)
-
-            bounding_boxes = tf.vectorized_map(
+            bounding_boxes = bounding_box.to_dense(bounding_boxes)
+            bounding_boxes = tf.map_fn(
```

Comment:
Why changing from vectorized_map to map_fn?

Reply:
map_fn is able to output a different output structure than input structure using fn_output_signature, whereas vectorized map does not expose an argument to specify output signature.  

So this is required to output the dictionary format.